swift-extensions/UIButtonExtensions.swift

```diff
@@ -69,11 +69,10 @@ extension UIButton {
                     self?.updateBackgroundColor(selector)
                 }
 
-                let imageResourceCombineLatest = CombineLatest.Companion().combine2(pub1: buttonViewModel.imageResource, pub2: buttonViewModel.tintColor)
-                observe(imageResourceCombineLatest) { [weak self] (result: CombineLatestResult2<StateSelector<ImageResource>, StateSelector<Color>>) in
-                    guard let imageSelector = result.component1() as? StateSelector<ImageResource>,
-                        let tintSelector = result.component2() as? StateSelector<Color>
-                        else { return }
+                let imageResourceCombineLatest = CombineLatestProcessorExtensionsKt.combine(buttonViewModel.imageResource, publishers: [buttonViewModel.tintColor])
+                observe(imageResourceCombineLatest) { [weak self] (value: [Any?]) in
+                    guard let imageSelector = value[0] as? StateSelector<ImageResource>,
+                        let tintSelector = value[1] as? StateSelector<Color> else { return }
```

Comment:
Is there any risk by accessing these via index on the array?

Reply:
Nope, since the combine contains a static number (2) of items